src/test/html/html-document_test.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {assert} from 'chai';
+import * as dom5 from 'dom5';
+import * as fs from 'fs';
+import * as path from 'path';
+import {HtmlParser} from '../../html/html-parser';
+import {ParsedHtmlDocument} from '../../html/html-document';
+
+suite('ParsedHtmlDocument', () => {
+
+  suite('sourceRangeForNode()', () => {
+    const parser: HtmlParser = new HtmlParser();
+    const url = '/static/source-ranges/html-complicated.html';
+    const file = fs.readFileSync(
+        path.resolve(__dirname, `../${url}`), 'utf8');
+    const document: ParsedHtmlDocument = parser.parse(file, url);
+
+    test.skip('can report correct range for comments', () => {});
+
+    test.skip('can report correct range for elements', () => {
+      // TODO(usergenic): If element source range includes their children,
+      // shouldn't they extend to include their closing tag as well?  They
+      // don't do so currently.  Add test after we discuss/and/or fix.
```

Comment:
Makes sense that if you want the entire element to be squiggled then that should include the end tag if any.